interface/reports/reports_controllers/ApptEncounterController.php

```diff
@@ -0,0 +1,316 @@
+	echo "  </td>\n";
+	echo "  <td align='right'>\n";
+	echo "   &nbsp;"; bucks($docrow['charges']); echo "&nbsp;\n";
+	echo "  </td>\n";
+	echo "  <td align='right'>\n";
+	echo "   &nbsp;"; bucks($docrow['copays']); echo "&nbsp;\n";
+	echo "  </td>\n";
+	echo "  <td colspan='2'>\n";
+	echo "   &nbsp;\n";
+	echo "  </td>\n";
+	echo " </tr>\n";
+
+	$grand_total_charges     += $docrow['charges'];
+	$grand_total_copays      += $docrow['copays'];
+	$grand_total_encounters  += $docrow['encounters'];
+
+	$docrow['charges']     = 0;
+	$docrow['copays']      = 0;
+	$docrow['encounters']  = 0;
+
+	$totals = array('grand_total_charges' => $grand_total_charges,
+	  				'grand_total_copays' => $grand_total_copays,
+	  				'grand_total_encounters' => $grand_total_encounters);
+	return $totals;
+}
+
+/*
+ * This function is for displaying the appointment and encou
+ * @params None
+ * @return void - Simply echo HTML encoded string
+ * @author: Tigpezeghe Rodrige
+ */
+function displayResult($res) {
+	$docrow = array('docname' => '', 'charges' => 0, 'copays' => 0, 'encounters' => 0);
+
+              while ($row = sqlFetchArray($res)) { // start of row
+                $patient_id = $row['pid'];
+                $encounter  = $row['encounter'];
+                $docname    = $row['docname'] ? $row['docname'] : xl('Unknown');
+
+                if ($docname != $docrow['docname']) {
+                  endDoctor($docrow);
+                }
+
+                $errmsg  = "";
+                $billed  = "Y";
+                $charges = 0;
+                $copays  = 0;
+                $gcac_related_visit = false;
+
+                // Scan the billing items for status and fee total.
+                //
+                $query = "SELECT code_type, code, modifier, authorized, billed, fee, justify " .
+                  "FROM billing WHERE " .
+                  "pid = '$patient_id' AND encounter = '$encounter' AND activity = 1";
+                $bres = sqlStatement($query);
+
+                while ($brow = sqlFetchArray($bres)) {
+                  $code_type = $brow['code_type'];
+                  if ($code_types[$code_type]['fee'] && !$brow['billed'])
+                    $billed = "";
+                  if (!$GLOBALS['simplified_demographics'] && !$brow['authorized'])
+                    postError(xl('Needs Auth'));
+                  if ($code_types[$code_type]['just']) {
+                    if (! $brow['justify']) postError(xl('Needs Justify'));
+                  }
+                  if ($code_types[$code_type]['fee']) {
+                    $charges += $brow['fee'];
+                    if ($brow['fee'] == 0 && !$GLOBALS['ippf_specific']) postError(xl('Missing Fee'));
+                  } else {
+                    if ($brow['fee'] != 0) postError(xl('Fee is not allowed'));
+                  }
+
+                  // Custom logic for IPPF to determine if a GCAC issue applies.
+                  if ($GLOBALS['ippf_specific']) {
```

Comment:
ippf_specific is not a in-use global.

Reply:
Should I remove the ippf_specific condition around that block?

Reply:
I vote to remove the ippf stuff from the entire code base. May need an issue and a plan.